enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+
+- Deployments for InstallStrategy
+- RBAC w/ subjects matching Deployment ServiceAccounts for Permissions and Owned, Required, and NativeAPIs
+- APIServices and Services for APIServiceDefinitions
+
+Required, provided, optional, and native APIs in a `k8s+v1` bundle will inferred by inspecting the RBAC, CRDs, Deployments, and APIServices it contains using the following rules:
+
+- the GVK of any CRD in the bundle will be considered a **provided API**
+- a GVK referenced in RBAC with a subject shared by the Deployment will be considered:
+  - **optional** if the referencing `(Cluster)RoleBinding` has the `operators.operatorframework.io.bundle.dependency="optional"` annotation set
+    - i.e. the GVK is neither provided or required
+  - **provided** if it belongs to the group version of an APIService in the bundle
+  - **required** if it DOES NOT belong to:
+    - any group version defined by an APIService or CRD in the bundle
+    - a `*.k8s.io` group
+- `nonResourceURLs` referenced in RBAC with a subject shared by the Deployment will be considered **native APIs**
+
+The dependence on explicit kinds in RBAC rules means that rules with wildcard types (i.e. `*`) are disallowed in `k8s+v1` bundles. This may seem like a burdensome restriction, but a nice corollary is that all `k8s+v1` bundles follow [the principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). The exception to the restriction is RBAC w/ subjects that do not match a ServiceAccount in the bundle's Deployments.
+
+#### Non-Inferrable
+
+The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+
+- Name
+- Descriptors
+- Description
+- Display Name
+- Keywords
+- Labels
+- Links
+- Maintainers
+- Maturity
+- Provider
+- Selector
+- Icon
+- Runtime Metadata
+- MinKubeVersion
+- (Future) Optional APIs
+- Replaces
+- Version
+- InstallModes
+
+This non-inferrable data will be gathered through a mixture of resource annotations and an `olm.yaml` file in the `metadata` directory of the bundle.
+
+Ex. _olm.yaml_
+
+```yaml
+name: packageserver-v1.14.1
+maturity: alpha
+version: 0.14.1
+replaces: packageserver-v1.14.0
+displayName: Package Server
+description: Represents an Operator package that is available from a given CatalogSource which will resolve to a ClusterServiceVersion.
+minKubeVersion: 1.11.0
+keywords: ['packagemanifests', 'olm', 'packages']
+maintainers:
+- name: Red Hat
+email: openshift-operators@redhat.com
+provider:
+name: Red Hat
+links:
+- name: Package Server
+url: https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/package-server
+installModes:
+- type: OwnNamespace
+  supported: true
+- type: SingleNamespace
+  supported: true
+- type: MultiNamespace
+  supported: true
+- type: AllNamespaces
+  supported: true
+icon: my-icon-png-base64-encoded
```

Comment:
This should rather point to a file in the same directory as `olm.yml` instead of an inlined base64-encoded string.

Reply:
Or perhaps a URL (which could support schemes for bundle-local file paths as well as the existing string, and leave room for extension later)?

Reply:
A URL would probably invite too many people using that, which would break disconnected environments and potentially open an attack vector when referencing external, non-trusted content without any checks.

Reply:
That wouldn't be a problem with URLs per se, but it's a good point: there may not exist any schemes other than bundle-local file paths that are useful and safe.